### PR TITLE
chore: improves using of Spring Security

### PR DESCRIPTION
**Pull request by jekutzsche**
Now uses the common way of doing authentication via a controller. The `UsernamePasswordAuthenticationFilter` assumes inappropriate preconditions and makes the authentication more complicated than necessary.

JWT authentication is now done via Spring-Security's OAuth2 integration rather than a custom filter.

The packages `users` and `auth.db` are now more cleanly separated in their responsibilities.

### CODE REVIEW

File: iris-client-bff/src/main/java/iris/client_bff/auth/db/web/LoginController.java [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+		var token = jwtService.createToken(user);
+
+		var headers = new HttpHeaders();
+		headers.add(AUTHENTICATION_INFO, BEARER_TOKEN_PREFIX + token);
+		headers.add(HttpHeaders.ACCESS_CONTROL_EXPOSE_HEADERS, AUTHENTICATION_INFO);
+
+		return ResponseEntity.ok().headers(headers).body(new LoginResponseDto(token));
+	}
+
+	// @ExceptionHandler
```
Comment (mad-nuts) [inline iris-client-bff/src/main/java/iris/client_bff/auth/db/web/LoginController.java:78]: Can this be removed / cleaned up?

**Review by mad-nuts — APPROVED** [review]

**Review by lucky-lusa — APPROVED** [review]
LGTM

**Pull request merged**